webcurator-core/src/main/java/org/webcurator/core/harvester/coordinator/HarvestAgentManagerImpl.java

```diff
@@ -78,13 +78,15 @@ public void heartbeat(HarvestAgentStatusDTO aStatus) {
                 log.debug("Heartbeat for ti: {}, state: {}", tiOid, harvesterStatus.getStatus());
 
                 String harvesterStatusValue = harvesterStatus.getStatus();
-                if (StringUtils.isEmpty(harvesterStatusValue)) {
+                if (StringUtils.isEmpty(harvesterStatusDto.getHarvesterState()) || StringUtils.isEmpty(harvesterStatusValue) || harvesterStatusDto.getHarvesterState().equals("H3 Job Not Found")) {
                     log.error("harvesterStatusValue is null, tiOid:{}", tiOid);
+                    doHeartbeatLaunchFailed(ti, harvestResultNumber);
                     return;
                 }
 
                 if (harvesterStatusValue.startsWith("Paused")) {
                     doHeartbeatPaused(ti, harvestResultNumber);
+                    return;
```

Comment:
Should these new _return_ statements be _breaks_ instead? Otherwise it won't iterate through the rest of the harvests?

Reply:
Yes. It's incorret to return from a loop. I'll change it to "break" instead.

Reply:
By the way, I tested changing them to _break_ earlier and recompiled, and it didn't fix the blank stats issue